script/build-example-site.js

```diff
@@ -0,0 +1,22 @@
+var marked = require('marked');
+var fs = require('fs');
+var path = require('path');
+var rimraf = require('rimraf');
+
+var basePath = path.resolve(__dirname, '../') + '/';
+
+try {
+  rimraf.sync(basePath + 'resource/example-site');
+  fs.mkdirSync(basePath + 'resource/example-site');
+} catch(e) {
+  console.log(e);
+}
+
+var template = fs.readFileSync(basePath + 'resource/example-site.template.html', {encoding: 'utf-8'});
+var readmeText = fs.readFileSync(basePath + 'README.md', {encoding: 'utf-8'});
+
+var exampleSiteHtml = template
+  .replace('<!-- REPLACE -->', marked(readmeText))
+  .replace('src="resource', 'src="https://raw.githubusercontent.com/james-proxy/james/master/resource');
+
+fs.writeFileSync(basePath + 'resource/example-site/index.html', exampleSiteHtml);
```

Comment:
It's still a little weird to have temp files in resources, no?

If possible, though, it would be awesome to solve it without using gulp
On Feb 5, 2016 9:47 PM, "Tomi" <notifications@github.com> wrote:

> In script/build-example-site.js
> <https://github.com/james-proxy/james/pull/77#discussion_r52070526>:
>
> > +
> > +try {
> > +  rimraf.sync(basePath + 'resource/example-site');
> > +  fs.mkdirSync(basePath + 'resource/example-site');
> > +} catch(e) {
> > +  console.log(e);
> > +}
> > +
> > +var template = fs.readFileSync(basePath + 'resource/example-site.template.html', {encoding: 'utf-8'});
> > +var readmeText = fs.readFileSync(basePath + 'README.md', {encoding: 'utf-8'});
> > +
> > +var exampleSiteHtml = template
> > +  .replace('<!-- REPLACE -->', marked(readmeText))
> > +  .replace('src="resource', 'src="https://raw.githubusercontent.com/james-proxy/james/master/resource');
> > +
> > +fs.writeFileSync(basePath + 'resource/example-site/index.html', exampleSiteHtml);
>
> Is gulp really the right tool for that? I'd suggest defaulting to npm
> scripts and using gulp when you really need to.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/james-proxy/james/pull/77/files#r52070526>.
>